lib/Sema/DerivedConformanceCodingKey.cpp

```diff
@@ -405,8 +405,9 @@ static bool canSynthesizeCodingKey(TypeChecker &tc, Decl *parentDecl,
       return false;
   }
 
-  if (!enumDecl->getInherited().empty() &&
-      enumDecl->getInherited().front().isError())
+  auto inherited = enumDecl->getInherited();
+  if (!inherited.empty() && inherited.front().wasValidated() &&
```

Comment:
It's not the raw value type that needs to be validated, but the enum itself. Validating the enum will visit the items in the inheritance clause.